### PR TITLE
fix VPU square root corner case

### PR DESCRIPTION
**Pull request by Allan-xmos**

### CODE REVIEW

File: tests/bfp_tests/src/bfp/real/test_bfp_sqrt_vect.c [hunk 1]
```diff
@@ -73,7 +73,14 @@ TEST(bfp_sqrt, bfp_s16_sqrt)
 
             double diff = exp_val - a_val;
 
-            TEST_ASSERT(  fabs(diff) <= ldexp(2, A.exp) );
+            #if defined(__VX4B__)
+            // we lose a bit of precision on VX4B due to the 15-bit VLMUL shift, so allow a tolerance of 3 LSBs
```
Comment (uvvpavel) [inline tests/bfp_tests/src/bfp/real/test_bfp_sqrt_vect.c]: I think this comment doesn't make sense with the way we use ldexp

File: tests/vect_tests/src/vect/test_vect_sqrt.c [hunk 2]
```diff
@@ -102,6 +105,46 @@ TEST(vect_sqrt, vect_s32_sqrt_prepare)
...
+        right_shift_t b_shr;
+        vect_s32_sqrt_prepare(&a_exp, &b_shr, b_exp, b_hr);
+
+        int32_t A;
+        vect_s32_sqrt(&A, &B, 1, b_shr, VECT_SQRT_S32_MAX_DEPTH);
+
+        double expected = sqrt(ldexp((double) B, b_exp));
+        double actual = ldexp((double) A, a_exp);
+
+        TEST_ASSERT( fabs(expected - actual) <= ldexp(2, a_exp) );
```
Comment (uvvpavel) [inline tests/vect_tests/src/vect/test_vect_sqrt.c]: why is it 2 here and in the next test?

**Review by uvvpavel — COMMENTED** [review]
Could you add an entry to the changelog as well please

File: tests/vect_tests/src/vect/test_vect_sqrt.c (hunk 2 above)
Comment (Allan-xmos) [reply tests/vect_tests/src/vect/test_vect_sqrt.c]: Good point, it can be ldexp(1.1, a_exp) for int16, and <1LSB for int32

**Review by uvvpavel — COMMENTED** [review]
All looks good, could you squash before merging please?

**Review by uvvpavel — APPROVED** [review]
sorry, actually approving
